IBM/carbon-addons-iot-react
# feat(tablecard): combined CardHeader with TableHeader #820

Closes #820 

New functionality Screenshot (See header):
![Screen Shot 2020-01-15 at 10 50 09 AM](https://user-images.githubusercontent.com/25177649/72453593-f42caa00-3784-11ea-9b39-1280ee38de50.png)

**Summary**

- Refactored TableCard to receive the CardToolbar as `customToolbarContent`. The `customToolbarContent` was moved to the right of the Table's actions.

- Added the ability to add a `secondaryTitle` and `tooltip` as a smaller header, which deprecates the current `hasRowCountInHeader` prop as secondaryTitle is more general.
  - Added deprecation notice to hasRowCountInHeader

- Removed the Toolbar and ToolbarItem from Card as they are deprecated and have no use

- Updated CardToolbar to use larger buttons to reflect the functionality that TableCard has, as well as enlarge the click-space

- Changed all images to use the same 20x20 size

- Updated affected tests to reflect new class names and components

- Updated snapshots to reflect new changes

- Updated stories to reflect new features

**Change List (commits, features, bugs, etc)**

- Components
  - src/components/Card/CardRangePicker.jsx
  - src/components/Card/CardToolbar.jsx
  - src/components/Table/Table.jsx
  - src/components/Table/TableToolbar/TableToolbar.jsx
  - src/components/TableCard/TableCard.jsx
  - src/constants/PropTypes.js
- Styles
  - src/components/Card/_card.scss
  - src/components/DataTable/_data-table.scss
- Tests
  - src/components/Card/Card.test.jsx
  - src/components/Table/Table.test.jsx
  - src/components/TableCard/TableCard.test.jsx
- Stories
  - src/components/Table/Table.story.jsx
  - src/components/TableCard/TableCard.story.jsx

**Acceptance Test (how to verify the PR)**

- Check Card story to ensure all buttons still function correctly and that the buttons' height expands the whole header
- Check TableCard story to ensure there is only 1 header shown with title, tableActions, and cardActions. Also ensure all of the buttons work correctly.
